rust/aptos-indexer-transaction-stream/src/transaction_stream.rs

```diff
@@ -582,42 +580,224 @@ pub async fn create_fetcher_loop(
+                    processor_name = self.processor_name,
+                    service_type = PROCESSOR_SERVICE_TYPE,
+                    stream_address = self.indexer_grpc_data_service_address.to_string(),
+                    connection_id = self.connection_id,
+                    start_version = self.starting_version,
+                    end_version = self.request_ending_version,
                     error = ?e,
                     "[Parser] Timeout receiving datastream response."
                 );
                 false
             },
         };
         // Check if we're at the end of the stream
-        let is_end = if let Some(ending_version) = request_ending_version {
-            next_version_to_fetch > ending_version
+        let is_end = if let Some(ending_version) = self.request_ending_version {
+            self.next_version_to_fetch > ending_version
         } else {
             false
         };
+
         if is_end {
             info!(
-                processor_name = processor_name,
-                service_type = crate::worker::PROCESSOR_SERVICE_TYPE,
-                stream_address = indexer_grpc_data_service_address.to_string(),
-                connection_id,
-                ending_version = request_ending_version,
-                next_version_to_fetch = next_version_to_fetch,
+                processor_name = self.processor_name,
+                service_type = PROCESSOR_SERVICE_TYPE,
+                stream_address = self.indexer_grpc_data_service_address.to_string(),
+                connection_id = self.connection_id,
+                ending_version = self.request_ending_version,
+                next_version_to_fetch = self.next_version_to_fetch,
                 "[Parser] Reached ending version.",
             );
+
+            TransactionStreamOutput {
+                transactions: transaction_pb_response,
+                should_continue_fetching: false,
+            }
+        } else {
+            // The rest is to see if we need to reconnect
+            if !is_success {
+                self.reconnect_to_grpc().await;
+            }
+
+            TransactionStreamOutput {
+                transactions: transaction_pb_response,
+                should_continue_fetching: true,
+            }
+        }
+    }
+
+    pub async fn reconnect_to_grpc(&mut self) {
+        // Sleep for 100ms between reconnect tries
+        // TODO: Turn this into exponential backoff
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+        if self.reconnection_retries >= RECONNECTION_MAX_RETRIES {
+            error!(
+                processor_name = self.processor_name,
+                service_type = PROCESSOR_SERVICE_TYPE,
+                stream_address = self.indexer_grpc_data_service_address.to_string(),
+                "[Parser] Reconnected more than 100 times. Will not retry.",
+            );
+            panic!("[Parser] Reconnected more than 100 times. Will not retry.")
```

Comment:
Same vibe here, I'd love if we didn't panic and instead returned a Result.

Reply:
Removed panics everywhere and replaced them with Result